eng/packages/http-client-csharp/generator/Azure.Generator/src/Visitors/DistributedTracingVisitor.cs

```diff
@@ -0,0 +1,275 @@
+
+        protected override ClientProvider? Visit(InputClient client, ClientProvider? clientProvider)
+        {
+            if (clientProvider == null)
+            {
+                return clientProvider;
+            }
+
+            AddDistributedTracingProperty(clientProvider);
+
+            return clientProvider;
+        }
+
+        protected override ConstructorProvider? VisitConstructor(ConstructorProvider constructor)
+        {
+            if (ShouldSkip(constructor.EnclosingType))
+            {
+                return constructor;
+            }
+
+            PropertyProvider clientDiagnosticsProperty = constructor.EnclosingType
+               .Properties.First(p => p.Name == ClientDiagnosticsPropertyName);
+            ParameterProvider? endpoint = constructor.Signature.Parameters.FirstOrDefault(p => p.Name == "endpoint");
+            ParameterProvider? options = constructor.Signature.Parameters.FirstOrDefault(p => p.Name == "options");
+            bool hasNoInitializer = constructor.Signature.Initializer == null;
+            bool isSubClientConstructor = hasNoInitializer &&
+                   constructor.Signature.Modifiers.HasFlag(MethodSignatureModifiers.Internal) &&
+                   endpoint != null;
+            bool isPrimaryConstructor = hasNoInitializer && options != null;
+
+            if (isSubClientConstructor)
+            {
+                var updatedSignature = new ConstructorSignature(
+                   constructor.Signature.Type,
+                   constructor.Signature.Description,
+                   constructor.Signature.Modifiers,
+                   [clientDiagnosticsProperty.AsParameter, .. constructor.Signature.Parameters],
+                   constructor.Signature.Attributes,
+                   constructor.Signature.Initializer);
+                var assignClientDiagnostics = clientDiagnosticsProperty.Assign(clientDiagnosticsProperty.AsParameter).Terminate();
+                List<MethodBodyStatement> updatedBody = constructor.BodyStatements != null
+                    ? [assignClientDiagnostics, constructor.BodyStatements]
+                    : [assignClientDiagnostics];
+
+                constructor.Update(signature: updatedSignature, bodyStatements: updatedBody);
+            }
+            else if (isPrimaryConstructor)
+            {
+                var assignClientDiagnostics = clientDiagnosticsProperty.Assign(
+                        New.Instance(clientDiagnosticsProperty.Type, [options!, True])).Terminate();
+                List<MethodBodyStatement> constructorBody = constructor.BodyStatements != null
+                    ? [constructor.BodyStatements, assignClientDiagnostics]
+                    : [assignClientDiagnostics];
+
+                constructor.Update(bodyStatements: constructorBody);
+            }
+
+            return constructor;
+        }
+
+        protected override MethodProvider? VisitMethod(MethodProvider method)
```

Comment:
I wonder if we can make this experience better when overriding ScmLibraryVisitor. Maybe ScmLibraryVisitor can override VisitMethod and then do the cast to ScmMethodProvider and call into a new overload `VisitMethod(ScmMethodProvider method)`